src/components/MessageInput/MessageInput.tsx

```diff
@@ -694,6 +654,10 @@ const areEqual = <
   const fileUploadsEqual = prevFileUploads.length === nextFileUploads.length;
   if (!fileUploadsEqual) return false;
 
+  const mentionedUsersEqual =
```

Comment:
We do a different check elsewhere for this:

```
  const mentionedUsersEqual =
    prevMessage.mentioned_users?.length ===
      nextMessage.mentioned_users?.length &&
    (nextMessage.mentioned_users?.length === 0 ||
      (prevMessage.mentioned_users?.length &&
        nextMessage.mentioned_users?.length &&
        prevMessage.mentioned_users[0].name ===
          nextMessage.mentioned_users[0].name));
  if (!mentionedUsersEqual) return false;
```

Reply:
Rest of the places the check is on message.mentioned_users, which is array of user objects. This one is array of user ids. And our mentioned_users functionality is a bit broken on input, since we never remove mentioned user (once it is added). So rest of the checks are not really necessary. We should update it when we fix mentioned users functionality itself.

Reply:
Basically right now, `mentioned_users` length only goes in one direction. We should add a ticket and fix it !!